src/Forms/quadraticform.jl

```diff
@@ -0,0 +1,70 @@
+struct QuadraticForm{V,KetMap,BraMap,OperatorMap,VectorSpaceMap,LinkSpaceMap} <:
+       AbstractForm{V,KetMap,BraMap,OperatorMap,VectorSpaceMap}
+  tn::ITensorNetwork{V}
+  bra_vs_map::BraMap
+  ket_vs_map::KetMap
+  operator_vs_map::OperatorMap
+  dual_space_map::VectorSpaceMap
+  link_space_map::LinkSpaceMap
+end
```

Comment:
One idea would be to have `QuadraticForm` wrap a `BilinearForm`, i.e.:
```julia
struct QuadraticForm{Form<:BilinearForm} <: AbstractForm
  form::Form
end

bilinear_form(qf::QuadraticForm) = qf.form
```
Then constructors and function calls could just forward to those of `BilinearForm`.

Reply:
I think that makes sense mathematically as well since `QuadraticForm` is a special case of `BilinearForm` (kind of like how in Julia's LinearAlgebra library a `Hermitian` matrix is a wrapper around a `Matrix`).

Reply:
Yeah I can see how this makes things cleaner and avoids the need for `construct_form.jl`